docs/tutorials/client-server.md

```diff
@@ -0,0 +1,230 @@
+
+:::workflow
+![Unique key-value pair](~/workflows/unique-kvp.bonsai)
+:::
+
+The [`DistinctBy`](xref:Bonsai.Reactive.DistinctBy) operator filters the output of [`Zip`](xref:Bonsai.Reactive.Zip) according to the unique `byte` value and produces a sequence containing only the distinct – or ‘new’ – values produced by [`Zip`](xref:Bonsai.Reactive.Zip). The output of [`DistinctBy`](xref:Bonsai.Reactive.DistinctBy) will therefore effectively be a sequence of unique client addresses corresponding to each connected client. We also need to store these unique values and make them available to other parts of the Bonsai workflow. 
+
+- Add a [`ReplaySubject`](xref:Bonsai.Reactive.ReplaySubject) operator after [`DistinctBy`](xref:Bonsai.Reactive.DistinctBy) and name it ‘ClientAddresses’. 
+
+:::workflow
+![Address ReplaySubject](~/workflows/address-replay-subject.bonsai)
+:::
+
+A [`ReplaySubject`](xref:Bonsai.Reactive.ReplaySubject) has the useful feature that it stores its input sequence and replays those values to any current or future subscribers. The effect in this case is that anything that subscribes to `ClientAddresses` will receive all the unique client addresses encountered by the server so far.
+
+## Server --> client communication
+Eventually, we will use these unique client addresses to route server messages back to specific client. For now, we'll implement a more basic approach where the server just sends messages back to the client that originally sent them. The Bonsai ZeroMQ library provides a convenient operator for this task in the form of [`SendResponse`](xref:Bonsai.ZeroMQ.SendResponse). 
+
+- Add a [`SendResponse`](xref:Bonsai.ZeroMQ.SendResponse) operator after the [`Router`](xref:Bonsai.ZeroMQ.Router) in a separate branch.
+- Inside (double-click on [`SendResponse`](xref:Bonsai.ZeroMQ.SendResponse)) add a [`String`](xref:Bonsai.Expressions.StringProperty) operator with a generic response value like `ServerResponse` after the `Source` operator. 
+
+:::workflow
+![Basic server response](~/workflows/server-basic-response.bonsai)
+:::
+
+The [`SendResponse`](xref:Bonsai.ZeroMQ.SendResponse) operator has a couple of interesting properties which may not be immediately obvious from this simple example. First, this operator always transmits its response back to the ZeroMQ socket that initiated the request (in this case one of our [`Dealer`](xref:Bonsai.ZeroMQ.Dealer) clients) and we therefore do not need to specify an address in its processing logic. Second, the internal flow of [`SendResponse`](xref:Bonsai.ZeroMQ.SendResponse) logic is computed asynchronously. This is very useful for responses that require more intensive computation and allows a [`Router`](xref:Bonsai.ZeroMQ.Router) to deal with frequent incoming [`Dealer`](xref:Bonsai.ZeroMQ.Dealer) requests efficiently. 
```

Comment:
I wonder if an analogy or connection with `SelectMany` should be made here, explaining that it is an operator that also runs in parallel for each request, but with an additional side-effect of sending the response back to the client.